docs/v3/core/v3.0.rst

```diff
@@ -363,814 +364,863 @@ terminology for a use case of reading from an array:
-    pairs, where the key must be a string and the value can be an arbitrary
-    JSON literal. Intended to allow storage of arbitrary user metadata.
+1. The encoded chunk representation ``EC_final`` is read from the store_.
 
-For example, the JSON document below defines an explicit group::
+2. If ``codecs[codecs.length]`` is a byte string, ``EC[codecs.length] :=
+   EC_final``.  Otherwise, ``EC[codecs.length]`` is
+   :ref:`converted<default-array-byte-string-conversion>` from ``EC_final``.
 
-    {
-        "zarr_format": 3,
-        "node_type": "group",
-        "attributes": {
-            "spam": "ham",
-            "eggs": 42,
-        }
-    }
+3. For each codec ``codecs[i]`` in ``codecs``, iterating in reverse order,
+   ``EC[i] := codecs[i].decode(EC[i+1], decoded_representation[i])``.
 
-The group metadata object must not contain any other names. Those are reserved
-for future versions of this specification. An implementation must fail to open
-zarr hierarchies or groups with unknown metadata fields, with the exception of
-objects with a ``"must_understand": false`` key-value pair.
+4. The chunk array ``A`` is equal to ``EC[0]``.
 
-.. note::
+Specifying codecs
+-----------------
 
-   A group does not need a metadata document to exist. (See implicit groups.)
+To allow for flexibility to define and implement new codecs, this
+specification does not define any codecs, nor restrict the set of
+codecs that may be used. Each codec must be defined via a separate
+specification. In order to refer to codecs in array metadata
+documents, each codec must have a unique identifier, which is a URI
+that dereferences to a human-readable specification of the codec. A
+codec specification must declare the codec identifier, and describe
+(or cite documents that describe) the encoding and decoding algorithms
+and the format of the encoded data.
+
+A codec may have configuration parameters which modify the behaviour
+of the codec in some way. For example, a compression codec may have a
+compression level parameter, which is an integer that affects the
+resulting compression ratio of the data. Configuration parameters must
+be declared in the codec specification, including a definition of how
+configuration parameters are represented as JSON.
 
+The Zarr core development team maintains a repository of codec
+specifications, which are hosted alongside this specification in the
+`zarr-specs GitHub repository`_, and which are
+published on the `zarr-specs documentation Web site
+<https://zarr-specs.readthedocs.io/>`_. For ease of discovery, it is
+recommended that codec specifications are contributed to the
+zarr-specs GitHub repository. However, codec specifications may be
+maintained by any group or organisation and published in any location
+on the Web. For further details of the process for contributing a
+codec specification to the zarr-specs GitHub repository, see the Zarr
+community process specification.
```

Comment:
Does the 'Zarr community process specification' mean [ZEP000](https://zarr.dev/zeps/active/ZEP0000.html) here?

Reply:
I just moved that text around, I didn't write it originally.  But I have now changed it to link to ZEP 0.

Reply:
Thanks!